pkgs/servers/sql/mysql/8.0.x.nix

```diff
@@ -0,0 +1,94 @@
+{ stdenv, fetchurl, cmake, bison
+, boost, libedit, libevent, lz4, ncurses, openssl, protobuf, readline, zlib, perl
+, pkgconfig
+, cctools, CoreServices, developer_cmds }:
+
+# Note: zlib is not required; MySQL can use an internal zlib.
+
+let
+self = stdenv.mkDerivation rec {
+  name = "mysql-${version}";
+  version = "8.0.16";
+
+  src = fetchurl {
+    url = "https://dev.mysql.com/get/Downloads/MySQL-8.0/mysql-8.0.16.tar.gz";
+    sha256 = "1spa6lkvgww4si0yjs0w4cpyr99gmmvvidsp53gbp2yw42cyi7wd";
+  };
+
+  patches = [
+    ./8.0-gc-sections.patch
+  ];
+
+  preConfigure = stdenv.lib.optional stdenv.isDarwin ''
+    ln -s /bin/ps $TMPDIR/ps
+    export PATH=$PATH:$TMPDIR
+  '';
+
+  nativeBuildInputs = [ cmake bison ];
+
+  buildInputs = [ boost libedit libevent lz4 ncurses openssl pkgconfig protobuf readline zlib ]
+     ++ stdenv.lib.optionals stdenv.isDarwin [ perl cctools CoreServices developer_cmds ];
+
+  enableParallelBuilding = true;
```

Comment:
Thanks for pointing this out. I had just copied it over from `5.7.x.nix`, but it's interesting because (when building 8.0.16 by hand) there were some discrepancies between single-threaded `make` (which I eventually got working) and multi-threaded `make -j2` (which I never got to work completely).

Regardless, since it's not needed, I've removed it.

There may be some connection with the choice of stdenv, so I'll put some more detail in the section about stdenv.